x-pack/legacy/plugins/lens/server/usage/task.ts

```diff
@@ -0,0 +1,235 @@
+          },
+        },
+      },
+    },
+  };
+
+  const metrics: AggregationSearchResponseWithTotalHitsAsInt<
+    unknown,
+    {
+      body: { aggs: typeof aggs };
+    }
+  > = await callCluster('search', {
+    index: kibanaIndex,
+    rest_total_hits_as_int: true,
+    body: {
+      query: {
+        bool: {
+          filter: [
+            { term: { type: 'lens-ui-telemetry' } },
+            { range: { 'lens-ui-telemetry.date': { gte: 'now-90d/d' } } },
+          ],
+        },
+      },
+      aggs,
+    },
+    size: 0,
+  });
+
+  const byDateByType: Record<string, Record<string, number>> = {};
+  const suggestionsByDate: Record<string, Record<string, number>> = {};
+
+  metrics.aggregations!.daily.buckets.forEach(daily => {
+    const byType: Record<string, number> = byDateByType[daily.key] || {};
+    daily.groups.buckets.regularEvents.names.buckets.forEach(bucket => {
+      byType[bucket.key] = bucket.sums.value || 0 + (byType[daily.key] || 0);
+    });
+    byDateByType[daily.key] = byType;
+
+    const suggestionsByType: Record<string, number> = suggestionsByDate[daily.key] || {};
+    daily.groups.buckets.suggestionEvents.names.buckets.forEach(bucket => {
+      suggestionsByType[bucket.key] = bucket.sums.value || 0 + (suggestionsByType[daily.key] || 0);
+    });
+    suggestionsByDate[daily.key] = suggestionsByType;
+  });
+
+  // Always delete old date because we don't report it
+  await callCluster('deleteByQuery', {
+    index: kibanaIndex,
+    waitForCompletion: true,
+    body: {
+      query: {
+        bool: {
+          filter: [
+            { term: { type: 'lens-ui-telemetry' } },
+            { range: { 'lens-ui-telemetry.date': { lt: 'now-90d/d' } } },
+          ],
+        },
+      },
+    },
+  });
+
+  return {
+    byDate: byDateByType,
+    suggestionsByDate,
+  };
+}
+
+export function telemetryTaskRunner(server: Server) {
+  return ({ taskInstance }: RunContext) => {
+    const { state } = taskInstance;
+    const callCluster = server.plugins.elasticsearch.getCluster('admin').callWithInternalUser;
+
+    return {
+      async run() {
+        const kibanaIndex = server.config().get<string>('kibana.index');
+
+        return Promise.all([
+          getDailyEvents(kibanaIndex, callCluster),
+          getVisualizationCounts(callCluster, server.config()),
+        ])
+          .then(([lensTelemetry, lensVisualizations]) => {
+            return {
+              state: {
+                runs: (state.runs || 0) + 1,
+                byDate: (lensTelemetry && lensTelemetry.byDate) || {},
+                suggestionsByDate: (lensTelemetry && lensTelemetry.suggestionsByDate) || {},
+                saved: lensVisualizations,
+              },
+              runAt: getNextMidnight(),
+            };
+          })
+          .catch(errMsg =>
+            server.log(['warning'], `Error executing lens telemetry task: ${errMsg}`)
+          );
+      },
+    };
+  };
+}
+
+function getNextMidnight() {
```

Comment:
You're using moment everywhere else, why not here?

Reply:
This was copied